lib/Collection/QueryType/Handler/TaxonHandler.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Netgen\Layouts\Sylius\Collection\QueryType\Handler;
+
+use Netgen\Layouts\API\Values\Collection\Query;
+use Netgen\Layouts\Collection\QueryType\QueryTypeHandlerInterface;
+use Netgen\Layouts\Parameters\ParameterBuilderInterface;
+use Netgen\Layouts\Parameters\ParameterType;
+use Netgen\Layouts\Sylius\Parameters\ParameterType as SyliusParameterType;
+use Sylius\Component\Taxonomy\Model\TaxonInterface;
+use Sylius\Component\Taxonomy\Repository\TaxonRepositoryInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+use function count;
+use function trim;
+
+class TaxonHandler implements QueryTypeHandlerInterface
+{
+    /**
+     * @param TaxonRepositoryInterface<TaxonInterface> $taxonRepository
+     */
+    public function __construct(
+        private readonly TaxonRepositoryInterface $taxonRepository,
+        private readonly RequestStack $requestStack,
+    ) {}
+
+    public function buildParameters(ParameterBuilderInterface $builder): void
+    {
+        $builder->add(
+            'use_current_taxon',
+            ParameterType\Compound\BooleanType::class,
+            [
+                'reverse' => true,
+            ],
+        );
+
+        $builder->get('use_current_taxon')->add(
+            'taxon_id',
+            SyliusParameterType\TaxonType::class,
+        );
+    }
+
+    /**
+     * @return iterable<array-key, TaxonInterface>
+     */
+    public function getValues(Query $query, int $offset = 0, ?int $limit = null): iterable
+    {
+        $currentRequest = $this->requestStack->getCurrentRequest();
+        if (!$currentRequest instanceof Request) {
```

Comment:
@igorkrz Why is this necessary?

Reply:
@RandyCupic To make PHPStan happy, because `getCurrentRequest` can return `null`.